tests/unit/s2n_connection_test.c

```diff
@@ -245,20 +245,22 @@ int main(int argc, char **argv)
         EXPECT_SUCCESS(s2n_connection_get_selected_digest_algorithm(conn, &output));
         EXPECT_EQUAL(S2N_TLS_HASH_NONE, output);
 
-        s2n_tls_hash_algorithm expected_output[] = {
-            S2N_TLS_HASH_NONE, S2N_TLS_HASH_MD5,
-            S2N_TLS_HASH_SHA1, S2N_TLS_HASH_SHA224,
-            S2N_TLS_HASH_SHA256, S2N_TLS_HASH_SHA384,
-            S2N_TLS_HASH_SHA512, S2N_TLS_HASH_MD5_SHA1,
-            S2N_TLS_HASH_NONE
+        s2n_tls_hash_algorithm expected_output[S2N_HASH_ALGS_COUNT] = {
+            [S2N_HASH_MD5] = S2N_TLS_HASH_MD5,
+            [S2N_HASH_SHA1] = S2N_TLS_HASH_SHA1,
+            [S2N_HASH_SHA224] = S2N_TLS_HASH_SHA224,
+            [S2N_HASH_SHA256] = S2N_TLS_HASH_SHA256,
+            [S2N_HASH_SHA384] = S2N_TLS_HASH_SHA384,
+            [S2N_HASH_SHA512] = S2N_TLS_HASH_SHA512,
+            [S2N_HASH_MD5_SHA1] = S2N_TLS_HASH_MD5_SHA1,
         };
 
         for (size_t i = S2N_TLS_HASH_NONE; i <= UINT16_MAX; i++) {
             struct s2n_signature_scheme test_scheme = *conn->handshake_params.client_cert_sig_scheme;
             test_scheme.hash_alg = i;
             conn->handshake_params.client_cert_sig_scheme = &test_scheme;
             conn->handshake_params.server_cert_sig_scheme = &test_scheme;
-            if (i <= S2N_HASH_ALGS_COUNT) {
+            if (i < S2N_HASH_ALGS_COUNT) {
                 EXPECT_SUCCESS(s2n_connection_get_selected_client_cert_digest_algorithm(conn, &output));
                 EXPECT_EQUAL(expected_output[i], output);
```

Comment:
Since `S2N_HASH_SHAKE256_64` wasn't added to the `expected_output`,  this result is equal since `s2n_signature_scheme_to_tls_iana` sets unknown signature schemes to `S2N_TLS_HASH_NONE`. Is this the correct behavior of this test? Would we want this test to fail when new hash algorithms are added to remind people to update `s2n_signature_scheme_to_tls_iana`?

Reply:
Oh nice catch. That's a potential issue with the signature version of this test too. I updated the switch case for both the hash and signature conversion methods to not use a "default" case, so the compiler will force us to add new options. It's a bit out of scope of this PR, but I can always move it to a separate PR.

Updating the tests (without just duplicating the switch statement lol) would be trickier. For the hash enum we might be able to assert on the count value, but the signature enum isn't just 0, 1, 2, ..., size. It's a sparse array.